### DISCUSSION TITLE
yaml

### DISCUSSION THREAD

**Discussion by jb-ssncse** [ultralytics/yolov5#2193] [General]
Hello,
Do i need to reset nc to custom dataset classes in models/ yolov5_.yaml while training apart from passing a separate .yaml file with nc, class names path to train, val and test as argument to train.py?

**Comment by glenn-jocher** [comment]
@jb-ssncse no you don't need to take any action here, YOLOv5 automatically uses the dataset class count when building a new model or output layer if there is a discrepancy between the data yaml and the model yaml/weights.

**Comment by jb-ssncse** [comment]
Thank you
